src/sentry/snuba/spans_rpc.py

```diff
@@ -215,6 +218,33 @@ def run_timeseries_query(
             granularity_secs,
             ["time"],
         )
+
+    if comparison_delta:
+        if len(rpc_request.aggregations) != 1:
+            raise InvalidSearchQuery("Only one column can be selected for comparison queries")
+
+        comp_query_params = params.copy()
+        assert comp_query_params.start is not None, "start is required"
+        assert comp_query_params.end is not None, "end is required"
+        comp_query_params.start -= comparison_delta
+        comp_query_params.end -= comparison_delta
```

Comment:
🤔 actually not sure if i like this more (or if mypy even likes it)
```suggestion
        comp_query_params.start = comp_query_params.start_date - comparison_delta
        comp_query_params.end = comp_query_params.end_date - comparison_delta
```